### ISSUE TITLE
Javadoc syntax highlighting doesn't work in JS files

### ISSUE THREAD

**Issue by 3Nex** [sublimehq/Packages#618]
Javadoc syntax highlighting works in PHP files for example, highlighting keywords such as @param, @return etc. 
![javadoc_highlighting](https://cloud.githubusercontent.com/assets/851674/18429215/a13c10c0-78d2-11e6-97e1-cdd994dd761d.png)

But none of it works in Javascript files. 

(I'm not sure if there's a valid reason behind it maybe?)

**Comment by wbond** [comment]
This is likely because no one has implemented it.

Further, I think that JS has not standardized on JavaDoc syntax like PHP has.

I'd be interested in getting info from the JS community about documentation standards before taking the time to implement this.

**Comment by 3Nex** [comment]
Originally I just wanted to know if it's in any way configurable so that I could get that to work at least just for myself. But I was advised on the Sublimetext forum to report it as an issue. 
https://forum.sublimetext.com/t/can-i-get-javadoc-syntax-highlighting-in-js-files/22762

Is it possible to just add the `.js` file extension somewhere in some config file to get it to work?

**Comment by wbond** [comment]
No, syntax highlighting isn't a case of configuration, but rather writing regex rules as part of the syntax itself. There isn't a way to turn on or off parts of highlighting via settings at the moment.

I think the way forward for this would be to look up info about the various JS block comment syntax modes to try and see which would be the best for us to support. Or perhaps we can find that multiple can be supported at the same time since the syntax does not conflict.

**Comment by FichteFoll** [comment]
I think the most commonly used standard (if there even are multiple) is jsdoc: http://usejsdoc.org/

I'm not a webdev, but I did use that the last time I was involved with it.

**Comment by Thom1729** [comment]
@3Nex If you're still interested in this functionality, open an issue at https://github.com/Thom1729/Sublime-JS-Custom/issues and I'll see what I can do.

**Comment by jingyuexing** [comment]
> 
> 
> Javadoc syntax highlighting works in PHP files for example, highlighting keywords such as `@param`, `@return` etc.
> 
> But none of it works in Javascript files.
> 
> (I'm not sure if there's a valid reason behind it maybe?)
> 
> ![javadoc_highlighting](https://cloud.githubusercontent.com/assets/851674/18429215/a13c10c0-78d2-11e6-97e1-cdd994dd761d.png)

user Vscode syntax defined file  transformation to sublime syntax
## like this :
![图片](https://user-images.githubusercontent.com/19589872/89881690-6a4c9e80-dbf8-11ea-99a7-b74fe6f72d8d.png)
![图片](https://user-images.githubusercontent.com/19589872/89881723-746e9d00-dbf8-11ea-88aa-c710f3823daa.png)
 use packdev
![图片](https://user-images.githubusercontent.com/19589872/89881831-96681f80-dbf8-11ea-8fe3-761862cd52b1.png)

vscode syntax you can in path `VSCode\resources\app\extensions` find it

**Comment by jingyuexing** [comment]
> 
> 
> Javadoc syntax highlighting works in PHP files for example, highlighting keywords such as `@param`, `@return` etc.
> 
> But none of it works in Javascript files.
> 
> (I'm not sure if there's a valid reason behind it maybe?)
> 
> ![javadoc_highlighting](https://cloud.githubusercontent.com/assets/851674/18429215/a13c10c0-78d2-11e6-97e1-cdd994dd761d.png)

first transfrom to plist,next transfrom to sublime-syntax

**Comment by rchl** [comment]
I'd also say that JSDoc is the prevalent documentation format in JS. Maybe especially due to Typescript [supporting a subset of it](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html).

**Comment by wbond** [comment]
If someone is interested in writing JSDoc support for the JavaScript syntax, I’d review it for inclusion.

**Comment by rchl** [comment]
I've started looking at it yesterday but since I have no previous experiences with syntaxes I can't promise that I'll get somewhere and if I do, that it will be of good quality. :)
